lib/utils.js

```diff
@@ -88,17 +88,20 @@ utils.setHttpAgent = function setHttpAgent(
   bypassProxy = false,
   AwsConstructor
 ) {
-  config.httpOptions = _.merge(config.httpOptions, {});
-
-  if (bypassProxy) {
-    config.httpOptions.agent = new https.Agent({
+  const getNonProxyAgent = () =>
+    new https.Agent({
       keepAlive: true,
       rejectUnauthorized: true,
       maxSockets: 50
     });
-  }
 
-  config = _.omit(config, 'bypassProxy');
```

Comment:
any idea why this might have been here before? i can't seem to find a reason for it, but want to make sure that we're not removing something we need.